clients/golang/pkg/client/client.go

```diff
@@ -0,0 +1,162 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"taskforce.sh/bullmq_proxy_client/pkg/client/proxyapi"
+	"taskforce.sh/bullmq_proxy_client/pkg/client/wsclient"
+)
+
+// client represents a rooturl
+type Client struct {
+	rootUrl   string
+	authToken string
+
+	// non configurable below / set on construction
+	httpClient *resty.Client
+}
+
+type ClientOpts func(*Client)
+
+func NewClient(opts ...ClientOpts) *Client {
+	c := &Client{
+		rootUrl:   "http://localhost:8080",
+		authToken: "",
+	}
+	for _, v := range opts {
+		v(c)
+	}
+
+	c.httpClient = resty.New().
+		SetBaseURL(c.rootUrl).
+		SetTimeout(1*time.Minute).
+		SetHeader("authorization", "Bearer "+c.authToken)
+
+	return c
+}
+
+func (c *Client) DialQueue(ctx context.Context, queueName string) (*wsclient.WebSocket[*proxyapi.QueueCommand], error) {
+	joinedPath, err := url.JoinPath(c.rootUrl, "ws", "queues", queueName)
+	if err != nil {
+		return nil, err
+	}
+	return wsclient.New[*proxyapi.QueueCommand](ctx, joinedPath, c.httpClient.Header)
+}
+
+func (c *Client) DialWorker(ctx context.Context, queueName string, concurrency int) (*wsclient.WebSocket[*proxyapi.WorkerCommand], error) {
+	joinedPath, err := url.JoinPath(c.rootUrl, "ws", "queues", queueName, "process", strconv.Itoa(concurrency))
+	if err != nil {
+		return nil, err
+	}
+	return wsclient.New[*proxyapi.WorkerCommand](ctx, joinedPath, c.httpClient.Header)
+}
+
+func (c *Client) DialQueueEvents(ctx context.Context, queueName string) (*wsclient.WebSocket[any], error) {
+	joinedPath, err := url.JoinPath(c.rootUrl, "ws", "queues", queueName, "events")
+	if err != nil {
+		return nil, err
+	}
+	return wsclient.New[any](ctx, joinedPath, c.httpClient.Header)
+}
+
+func (c *Client) AddJobs(ctx context.Context, queueName string, jobs []*proxyapi.JobSpec) (out []*proxyapi.Job, err error) {
+	_, err = c.httpClient.R().
+		SetBody(jobs).
+		SetResult(out).
+		ForceContentType("application/json").
```

Comment:
resty should be configured to use JSON by default so that it is not required to use ForceContentType here.

Reply:
i wasn't sure about this because some endpoints seem to return an `OK` instead of `"OK"`. since the first is not valid JSON i am worried.